packages/react-positioning/src/usePopper.ts

```diff
@@ -401,30 +231,48 @@ export function usePopper(
     if (options.target) {
       overrideTargetRef.current = options.target;
     }
-  }, [options.target, overrideTargetRef]);
-  useIsomorphicLayoutEffect(() => {
-    handlePopperUpdate();
+  }, [options.target, overrideTargetRef, containerRef]);
 
-    return () => {
-      popperInstanceRef.current?.destroy();
-      popperInstanceRef.current = null;
-    };
-  }, [handlePopperUpdate, options.enabled]);
+  useIsomorphicLayoutEffect(
+    () => {
+      updatePosition();
+    },
+    // Missing deps:
+    // updatePosition - referentially stable
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [enabled],
+  );
+
+  // Update position if rerender caused options to change
   useIsomorphicLayoutEffect(
     () => {
       if (!isFirstMount) {
-        popperInstanceRef.current?.setOptions(
-          resolvePopperOptions(overrideTargetRef.current ?? targetRef.current, containerRef.current, arrowRef.current),
-        );
+        updatePosition();
       }
     },
     // Missing deps:
     // isFirstMount - Should never change after mount
-    // arrowRef, containerRef, targetRef - Stable between renders
+    // updatePosition - Stable between renders
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [resolvePopperOptions],
```

Comment:
These effects look strange honestly, should it be just?

```ts
  useIsomorphicLayoutEffect(() => {
    updatePosition();
  }, [resolvePopperOptions, updatePosition]);
```

Before we were handling Popper.js instance, but it's not true anymore.

Reply:
yeah, but the lint rule still needs to be ignored since the lint rule does not recognize the callback refs as refs

Reply:
yeah but with the addition of `enabled` to the deps

Reply:
Done in [162ff51](https://github.com/microsoft/fluentui/pull/21828/commits/162ff51547fe04606b9ad235702b3d43021f0f68)